docker_tests/docker_tests_utils.py

```diff
@@ -28,7 +29,7 @@
 
 
 def run_command(cmd: List[str], print_output_on_error: bool = True, **kwargs):
-    print(f"$ {' '.join(c for c in cmd)}")
+    print(f"$ {' '.join(shlex.quote(c) for c in cmd)}")
```

Comment:
There’s `shlex.join()` on 3.8+; do we need to run this on older versions?

Reply:
I don't see any reason why we shouldn't support the old versions anymore. This is not a big change.

Reply:
Just to add a little more context on that one as well:

And to add on that - we do not know what python version the user has. The idea of this script (similarly as previous BASH scripts) is that it should run pretty much everywhere without any "extra" assumptions. Just pytho3 should be enough to run the script - I think we can - for now assume safely have `python3.6+` as the "big assumption".  

I would also love this to make sure it also works on Windows without WSL2 - this was the thing that was missing with `bash` approach - but ideally the workflow should be exactly the same for pure-windows developers too.

So we have to be extra careful this one also works for Windows (this is something that we will likely add during the Internship - and some of our interns are using Windows primarily - so this is something they will be able to check and likely we can even add tests for it in GitHub on Windows runners (but that's a bit later stage).

Reply:
Basically: with all those BashToPython changes we should assume that they are used by someone who has no dev env setup, nothing installed, just python (3.6+) installed on their OS (Modern Linux, MacOS, Windows). There should be no more prerequisites. Later we might add docker/docker-compose to the list of those prerequisites, but that should reall be it.

Reply:
Side - comment: Following the learnings from breeze implementation I am planning (this is also part of the internship work that we planned) to document all such assumptions in the format of ADRs https://adr.github.io/  so that we avoid the problem that a lot of decisions with Breeze did not have enough "context" explained.